src/main/java/ru/vk/itmo/viktorkorotkikh/io/write/AbstractSSTableWriter.java

```diff
@@ -0,0 +1,240 @@
+        } else {
+            blobBuffer = new ByteArraySegment(blockSize, false);
+        }
+    }
+
+    /**
+     * Writes entries to SSTable files including data, index, and compression information.
+     *
+     * @param isCompacted      A boolean flag indicating whether the SSTable is compacted.
+     * @param iteratorSupplier A supplier of iterators over the entries to be written.
+     * @param baseDir          The base directory for storing SSTable files.
+     * @param fileIndex        The index of the SSTable file.
+     * @throws IOException if an I/O error occurs during writing.
+     */
+    public void write(
+            boolean isCompacted,
+            Supplier<? extends Iterator<? extends Entry<MemorySegment>>> iteratorSupplier,
+            final Path baseDir,
+            final int fileIndex
+    ) throws IOException {
+        // Write to temporary files
+        final Path tempCompressionInfo = SSTable.tempCompressionInfoName(isCompacted, baseDir, fileIndex);
+        final Path tempIndexName = SSTable.tempIndexName(isCompacted, baseDir, fileIndex);
+        final Path tempDataName = SSTable.tempDataName(isCompacted, baseDir, fileIndex);
+
+        // Delete temporary files to eliminate tails
+        Files.deleteIfExists(tempIndexName);
+        Files.deleteIfExists(tempDataName);
+
+        // Iterate in a single pass!
+        // Will write through FileChannel despite extra memory copying and
+        // no buffering (which may be implemented later).
+        // Looking forward to MemorySegment facilities in FileChannel!
+        try (OutputStream compressionInfo =
+                     new BufferedOutputStream(
+                             new FileOutputStream(
+                                     tempCompressionInfo.toFile()),
+                             BUFFER_SIZE);
+             OutputStream index =
+                     new BufferedOutputStream(
+                             new FileOutputStream(
+                                     tempIndexName.toFile()),
+                             BUFFER_SIZE);
+             OutputStream data =
+                     new BufferedOutputStream(
+                             new FileOutputStream(
+                                     tempDataName.toFile()),
+                             BUFFER_SIZE)) {
+            index.write(new byte[(int) INDEX_METADATA_SIZE]); // write 0, fill in the data later
+
+            Iterator<? extends Entry<MemorySegment>> entries = iteratorSupplier.get();
+
+            writeCompressionHeader(compressionInfo);
+            compressionInfo.write(new byte[COMPRESSION_INFO_METADATA_SIZE]); // write 0s, fill in the data later
+
+            int entriesSize = 0;
+            boolean hasNoTombstones = true;
+
+            while (entries.hasNext()) {
+                // Then write the entry
+                final Entry<MemorySegment> entry = entries.next();
+                hasNoTombstones = entry.value() != null;
```

Comment:
Значение может многократно меняться между `true` и `false`, верно?
```suggestion
                hasNoTombstones &= entry.value() != null;
```

Reply:
Да, тут сам себя запутал. Ваше решение более корректное